packages/eslint-plugin/tests/rules/no-attribute-decorator.test.ts

```diff
@@ -0,0 +1,138 @@
+import rule, { RULE_NAME } from '../../src/rules/no-attribute-decorator';
+import { RuleTester } from '../test-helper';
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+});
+
+const messageId = 'noAttributeDecorator';
+
+ruleTester.run(RULE_NAME, rule, {
+  valid: [
+    // should pass if constructor does not exist
+    `
+      class Test {
+        public foo(){}
+      }
+    `,
+    // should pass if constructor exists but no parameter
+    `
+      class Test {
+        constructor() {}
+      }
+    `,
+    // should pass if constructor exists and have one parameter without decorator
+    `
+      class Test {
+        constructor(foo: any) {}
+      }
+    `,
+    // should pass if constructor exists and have one parameter with decorator
+    `
+      class Test {
+        constructor(@Optional() foo: any) {}
+      }
+    `,
+    // // should pass if constructor exists and have multiple parameters without decorator
+    `
+      class Test {
+        constructor(foo: any, @Optional() bar: any) {}
+      }
+    `,
+    // // should pass if constructor exists and have multiple parameters with decorator
+    `
+      class Test {
+        constructor(@Optional() foo: any, @Optional() bar: any) {}
+      }
+    `,
+  ],
+  invalid: [
+    {
+      // should fail if constructor has one parameter with @Attribute decorator
+      code: `
```

Comment:
Please use the `convertAnnotatedSourceToFailureCase()` utility to use the visual error annotations as has been done on the other rules

Reply:
Got it, I made the updates for the tests, please help review again. 
I suppose the `convertAnnotatedSourceToFailureCase` method only supports the scenario just have one error place, is that right?

Reply:
Maybe you mean something like this?
https://github.com/angular-eslint/angular-eslint/blob/master/packages/eslint-plugin-template/tests/rules/cyclomatic-complexity.test.ts#L120

Reply:
@paulotokimatu Cool, thanks for your information.

Reply:
@JamesHenry I have duplicated the method implementation of `convertAnnotatedSourceToFailureCase` from `eslint-plugin-template/tests` to `eslint-plugin/tests` to ensure the tests in the plugin folder also can take advantage of enhanced `convertAnnotatedSourceToFailureCase` method, but I'm wondering if we can extract this method to a shared folder so that no need to worry about the future refactoring of this method and duplicate it again.

Reply:
Good call out, thanks! I consolidated things into a utils package in #76, you can pull in that update and use the shared utils now

Reply:
@JamesHenry Got it, I pulled your commit and merge to my branch. I appreciate it if you can continue to review the PR again. 👍

Reply:
Hmm they still haven't been converted to use the multiple annotations though...